packages/grafana-runtime/src/utils/i18n.ts

```diff
@@ -0,0 +1,18 @@
+type UseTHook = () => (id: string, defaultMessage: string, values?: Record<string, unknown>) => string;
+
+// Fallback implementation that should be overridden by setUseT
```

Comment:
do we want some basic jsdoc here for the hook? e.g.
> hook that returns a translation function compatible with react-i18next

side note: given we control the name of this hook, we _could_ (not saying we should) give it a more descriptive name like `useTranslate()` or something, and just have the convention that you assign it to `t`. e.g. `const t = useTranslate()`. idk where i stand tbh, i kinda hate `useT` as a name, but equally it's closely tied to the react-i18next function so 🤷

Reply:
Yeah, I couldn't decide on a name. I think I agree that I prefer `useTranslate`. `useTranslations`? 🤔

Reply:
Pushed up `useTranslate`.